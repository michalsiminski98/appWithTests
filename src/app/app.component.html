<app-add-form></app-add-form>
<app-hide-button>SHOW / HIDE</app-hide-button>
<app-best-quote [title]= "titleForBest"></app-best-quote>
<app-quote-list [title] = "titleForList"></app-quote-list>
<app-worst-list [title] = "titleForWorst"></app-worst-list>