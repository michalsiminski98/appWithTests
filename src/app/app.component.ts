import { Component } from '@angular/core';

@Component({
  selector: 'app-root',
  templateUrl: './app.component.html',
  styleUrls: ['./app.component.css']
})
export class AppComponent {
  titleForBest: string = "Best Quote:"
  titleForList: string = "Quotes list:";
  titleForWorst: string = "Worst Quotes:";

  public isSubmitting: boolean = false;

  showForm = false;

  onShowForm(): void{
    this.showForm = !this.showForm;
  }

  adding(a:number, b:number): number{
    return a + b
  }
}

