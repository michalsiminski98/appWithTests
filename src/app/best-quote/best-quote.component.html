<ng-container *ngIf="bestQuote$ | async as bestQuote">
  <h2>{{title}}</h2>
<h3>{{bestQuote.author}}</h3>
<p>{{bestQuote.quotation}}</p>
<span>{{bestQuote.rate}}</span>
</ng-container>