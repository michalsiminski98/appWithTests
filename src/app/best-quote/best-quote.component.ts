import { Component, Input, OnInit } from '@angular/core';
import { Observable } from 'rxjs';
import { Quote } from '../interfaces/quote';
import { QuotesService } from '../services/quotes/quotes.service';

@Component({
  selector: 'app-best-quote',
  templateUrl: './best-quote.component.html',
  styleUrls: ['./best-quote.component.css']
})
export class BestQuoteComponent implements OnInit {

  @Input() title?: string;
  bestQuote$?: Observable<Quote>;

  constructor(private quotesService: QuotesService) { }

  ngOnInit(): void {
    this.bestQuote$ = this.quotesService.bestQuote$;
    

  }
}

