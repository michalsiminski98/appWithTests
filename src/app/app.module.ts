import { NgModule } from '@angular/core';
import { BrowserModule } from '@angular/platform-browser';

import { AppComponent } from './app.component';
import { HideButtonComponent } from './hide-button/hide-button.component';
import { AddFormComponent } from './add-form/add-form.component';
import { QuoteListComponent } from './quote-list/quote-list.component';
import { BestQuoteComponent } from './best-quote/best-quote.component';
import { WorstListComponent } from './worst-list/worst-list.component';
import { FormsModule } from '@angular/forms';

@NgModule({
  declarations: [
    AppComponent,
    HideButtonComponent,
    AddFormComponent,
    QuoteListComponent,
    BestQuoteComponent,
    WorstListComponent
  ],
  imports: [
    BrowserModule,
    FormsModule
  ],
  providers: [],
  bootstrap: [AppComponent]
})
export class AppModule { }
