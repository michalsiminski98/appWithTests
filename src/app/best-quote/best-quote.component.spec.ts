import { Shallow } from "shallow-render"
import { AppModule } from "../app.module";
import { BestQuoteComponent } from "./best-quote.component"

describe('bestQuoteComponent', () =>{
  let shallow: Shallow<BestQuoteComponent>;

  beforeEach(() => {
    shallow = new Shallow(
    BestQuoteComponent,
    AppModule
    )
  })

    it('should match snapshot', async () => {
      const{instance} = await shallow.render();
      expect(instance).toMatchSnapshot();
    })

})
