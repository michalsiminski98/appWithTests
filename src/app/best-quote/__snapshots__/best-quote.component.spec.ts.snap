// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`bestQuoteComponent should match snapshot 1`] = `
BestQuoteComponent {
  "bestQuote$": Observable {
    "_isScalar": false,
    "operator": MapOperator {
      "project": [Function],
      "thisArg": undefined,
    },
    "source": Observable {
      "_isScalar": false,
      "source": BehaviorSubject {
        "_isScalar": false,
        "_value": Array [
          Object {
            "author": "bardzo",
            "id": 0,
            "quotation": "asddasdasads adsasd a dsads  a ddsa asd.",
            "rate": -3,
          },
          Object {
            "author": "madry",
            "id": 1,
            "quotation": "AEDRFDHSRHTSHDERTaSER EAWGAERWEWF WEEF.",
            "rate": 3,
          },
          Object {
            "author": "autor",
            "id": 2,
            "quotation": "039284120934 012 4012981284 0914 29810 928.",
            "rate": 4,
          },
          Object {
            "author": "asgwfewefwd",
            "id": 3,
            "quotation": "9832 jhfp934h4r2  309 r48h23  984rh3  q98 4rhq329r3h4r98 h23r908q2h3r98h98h.",
            "rate": -5,
          },
        ],
        "closed": false,
        "hasError": false,
        "isStopped": false,
        "observers": Array [
          MapSubscriber {
            "_parentOrParents": Subscriber {
              "_parentOrParents": null,
              "_subscriptions": Array [
                [Circular],
              ],
              "closed": false,
              "destination": SafeSubscriber {
                "_complete": undefined,
                "_context": Object {
                  "unsubscribe": [Function],
                },
                "_error": [Function],
                "_next": [Function],
                "_parentOrParents": null,
                "_parentSubscriber": [Circular],
                "_subscriptions": null,
                "closed": false,
                "destination": Object {
                  "closed": true,
                  "complete": [Function],
                  "error": [Function],
                  "next": [Function],
                },
                "isStopped": false,
                "syncErrorThrowable": false,
                "syncErrorThrown": false,
                "syncErrorValue": null,
              },
              "isStopped": false,
              "syncErrorThrowable": true,
              "syncErrorThrown": false,
              "syncErrorValue": null,
            },
            "_subscriptions": Array [
              SubjectSubscription {
                "_parentOrParents": [Circular],
                "_subscriptions": null,
                "closed": false,
                "subject": [Circular],
                "subscriber": [Circular],
              },
            ],
            "closed": false,
            "count": 1,
            "destination": Subscriber {
              "_parentOrParents": null,
              "_subscriptions": Array [
                [Circular],
              ],
              "closed": false,
              "destination": SafeSubscriber {
                "_complete": undefined,
                "_context": Object {
                  "unsubscribe": [Function],
                },
                "_error": [Function],
                "_next": [Function],
                "_parentOrParents": null,
                "_parentSubscriber": [Circular],
                "_subscriptions": null,
                "closed": false,
                "destination": Object {
                  "closed": true,
                  "complete": [Function],
                  "error": [Function],
                  "next": [Function],
                },
                "isStopped": false,
                "syncErrorThrowable": false,
                "syncErrorThrown": false,
                "syncErrorValue": null,
              },
              "isStopped": false,
              "syncErrorThrowable": true,
              "syncErrorThrown": false,
              "syncErrorValue": null,
            },
            "isStopped": false,
            "project": [Function],
            "syncErrorThrowable": true,
            "syncErrorThrown": false,
            "syncErrorValue": null,
            "thisArg": [Circular],
          },
        ],
        "thrownError": null,
      },
    },
  },
  "quotesService": QuotesService {
    "_quotes": BehaviorSubject {
      "_isScalar": false,
      "_value": Array [
        Object {
          "author": "bardzo",
          "id": 0,
          "quotation": "asddasdasads adsasd a dsads  a ddsa asd.",
          "rate": -3,
        },
        Object {
          "author": "madry",
          "id": 1,
          "quotation": "AEDRFDHSRHTSHDERTaSER EAWGAERWEWF WEEF.",
          "rate": 3,
        },
        Object {
          "author": "autor",
          "id": 2,
          "quotation": "039284120934 012 4012981284 0914 29810 928.",
          "rate": 4,
        },
        Object {
          "author": "asgwfewefwd",
          "id": 3,
          "quotation": "9832 jhfp934h4r2  309 r48h23  984rh3  q98 4rhq329r3h4r98 h23r908q2h3r98h98h.",
          "rate": -5,
        },
      ],
      "closed": false,
      "hasError": false,
      "isStopped": false,
      "observers": Array [
        MapSubscriber {
          "_parentOrParents": Subscriber {
            "_parentOrParents": null,
            "_subscriptions": Array [
              [Circular],
            ],
            "closed": false,
            "destination": SafeSubscriber {
              "_complete": undefined,
              "_context": Object {
                "unsubscribe": [Function],
              },
              "_error": [Function],
              "_next": [Function],
              "_parentOrParents": null,
              "_parentSubscriber": [Circular],
              "_subscriptions": null,
              "closed": false,
              "destination": Object {
                "closed": true,
                "complete": [Function],
                "error": [Function],
                "next": [Function],
              },
              "isStopped": false,
              "syncErrorThrowable": false,
              "syncErrorThrown": false,
              "syncErrorValue": null,
            },
            "isStopped": false,
            "syncErrorThrowable": true,
            "syncErrorThrown": false,
            "syncErrorValue": null,
          },
          "_subscriptions": Array [
            SubjectSubscription {
              "_parentOrParents": [Circular],
              "_subscriptions": null,
              "closed": false,
              "subject": [Circular],
              "subscriber": [Circular],
            },
          ],
          "closed": false,
          "count": 1,
          "destination": Subscriber {
            "_parentOrParents": null,
            "_subscriptions": Array [
              [Circular],
            ],
            "closed": false,
            "destination": SafeSubscriber {
              "_complete": undefined,
              "_context": Object {
                "unsubscribe": [Function],
              },
              "_error": [Function],
              "_next": [Function],
              "_parentOrParents": null,
              "_parentSubscriber": [Circular],
              "_subscriptions": null,
              "closed": false,
              "destination": Object {
                "closed": true,
                "complete": [Function],
                "error": [Function],
                "next": [Function],
              },
              "isStopped": false,
              "syncErrorThrowable": false,
              "syncErrorThrown": false,
              "syncErrorValue": null,
            },
            "isStopped": false,
            "syncErrorThrowable": true,
            "syncErrorThrown": false,
            "syncErrorValue": null,
          },
          "isStopped": false,
          "project": [Function],
          "syncErrorThrowable": true,
          "syncErrorThrown": false,
          "syncErrorValue": null,
          "thisArg": [Circular],
        },
      ],
      "thrownError": null,
    },
    "bestQuote$": Observable {
      "_isScalar": false,
      "operator": MapOperator {
        "project": [Function],
        "thisArg": undefined,
      },
      "source": Observable {
        "_isScalar": false,
        "source": BehaviorSubject {
          "_isScalar": false,
          "_value": Array [
            Object {
              "author": "bardzo",
              "id": 0,
              "quotation": "asddasdasads adsasd a dsads  a ddsa asd.",
              "rate": -3,
            },
            Object {
              "author": "madry",
              "id": 1,
              "quotation": "AEDRFDHSRHTSHDERTaSER EAWGAERWEWF WEEF.",
              "rate": 3,
            },
            Object {
              "author": "autor",
              "id": 2,
              "quotation": "039284120934 012 4012981284 0914 29810 928.",
              "rate": 4,
            },
            Object {
              "author": "asgwfewefwd",
              "id": 3,
              "quotation": "9832 jhfp934h4r2  309 r48h23  984rh3  q98 4rhq329r3h4r98 h23r908q2h3r98h98h.",
              "rate": -5,
            },
          ],
          "closed": false,
          "hasError": false,
          "isStopped": false,
          "observers": Array [
            MapSubscriber {
              "_parentOrParents": Subscriber {
                "_parentOrParents": null,
                "_subscriptions": Array [
                  [Circular],
                ],
                "closed": false,
                "destination": SafeSubscriber {
                  "_complete": undefined,
                  "_context": Object {
                    "unsubscribe": [Function],
                  },
                  "_error": [Function],
                  "_next": [Function],
                  "_parentOrParents": null,
                  "_parentSubscriber": [Circular],
                  "_subscriptions": null,
                  "closed": false,
                  "destination": Object {
                    "closed": true,
                    "complete": [Function],
                    "error": [Function],
                    "next": [Function],
                  },
                  "isStopped": false,
                  "syncErrorThrowable": false,
                  "syncErrorThrown": false,
                  "syncErrorValue": null,
                },
                "isStopped": false,
                "syncErrorThrowable": true,
                "syncErrorThrown": false,
                "syncErrorValue": null,
              },
              "_subscriptions": Array [
                SubjectSubscription {
                  "_parentOrParents": [Circular],
                  "_subscriptions": null,
                  "closed": false,
                  "subject": [Circular],
                  "subscriber": [Circular],
                },
              ],
              "closed": false,
              "count": 1,
              "destination": Subscriber {
                "_parentOrParents": null,
                "_subscriptions": Array [
                  [Circular],
                ],
                "closed": false,
                "destination": SafeSubscriber {
                  "_complete": undefined,
                  "_context": Object {
                    "unsubscribe": [Function],
                  },
                  "_error": [Function],
                  "_next": [Function],
                  "_parentOrParents": null,
                  "_parentSubscriber": [Circular],
                  "_subscriptions": null,
                  "closed": false,
                  "destination": Object {
                    "closed": true,
                    "complete": [Function],
                    "error": [Function],
                    "next": [Function],
                  },
                  "isStopped": false,
                  "syncErrorThrowable": false,
                  "syncErrorThrown": false,
                  "syncErrorValue": null,
                },
                "isStopped": false,
                "syncErrorThrowable": true,
                "syncErrorThrown": false,
                "syncErrorValue": null,
              },
              "isStopped": false,
              "project": [Function],
              "syncErrorThrowable": true,
              "syncErrorThrown": false,
              "syncErrorValue": null,
              "thisArg": [Circular],
            },
          ],
          "thrownError": null,
        },
      },
    },
    "quotes": Array [
      Object {
        "author": "bardzo",
        "id": 0,
        "quotation": "asddasdasads adsasd a dsads  a ddsa asd.",
        "rate": -3,
      },
      Object {
        "author": "madry",
        "id": 1,
        "quotation": "AEDRFDHSRHTSHDERTaSER EAWGAERWEWF WEEF.",
        "rate": 3,
      },
      Object {
        "author": "autor",
        "id": 2,
        "quotation": "039284120934 012 4012981284 0914 29810 928.",
        "rate": 4,
      },
      Object {
        "author": "asgwfewefwd",
        "id": 3,
        "quotation": "9832 jhfp934h4r2  309 r48h23  984rh3  q98 4rhq329r3h4r98 h23r908q2h3r98h98h.",
        "rate": -5,
      },
    ],
    "quotes$": Observable {
      "_isScalar": false,
      "source": BehaviorSubject {
        "_isScalar": false,
        "_value": Array [
          Object {
            "author": "bardzo",
            "id": 0,
            "quotation": "asddasdasads adsasd a dsads  a ddsa asd.",
            "rate": -3,
          },
          Object {
            "author": "madry",
            "id": 1,
            "quotation": "AEDRFDHSRHTSHDERTaSER EAWGAERWEWF WEEF.",
            "rate": 3,
          },
          Object {
            "author": "autor",
            "id": 2,
            "quotation": "039284120934 012 4012981284 0914 29810 928.",
            "rate": 4,
          },
          Object {
            "author": "asgwfewefwd",
            "id": 3,
            "quotation": "9832 jhfp934h4r2  309 r48h23  984rh3  q98 4rhq329r3h4r98 h23r908q2h3r98h98h.",
            "rate": -5,
          },
        ],
        "closed": false,
        "hasError": false,
        "isStopped": false,
        "observers": Array [
          MapSubscriber {
            "_parentOrParents": Subscriber {
              "_parentOrParents": null,
              "_subscriptions": Array [
                [Circular],
              ],
              "closed": false,
              "destination": SafeSubscriber {
                "_complete": undefined,
                "_context": Object {
                  "unsubscribe": [Function],
                },
                "_error": [Function],
                "_next": [Function],
                "_parentOrParents": null,
                "_parentSubscriber": [Circular],
                "_subscriptions": null,
                "closed": false,
                "destination": Object {
                  "closed": true,
                  "complete": [Function],
                  "error": [Function],
                  "next": [Function],
                },
                "isStopped": false,
                "syncErrorThrowable": false,
                "syncErrorThrown": false,
                "syncErrorValue": null,
              },
              "isStopped": false,
              "syncErrorThrowable": true,
              "syncErrorThrown": false,
              "syncErrorValue": null,
            },
            "_subscriptions": Array [
              SubjectSubscription {
                "_parentOrParents": [Circular],
                "_subscriptions": null,
                "closed": false,
                "subject": [Circular],
                "subscriber": [Circular],
              },
            ],
            "closed": false,
            "count": 1,
            "destination": Subscriber {
              "_parentOrParents": null,
              "_subscriptions": Array [
                [Circular],
              ],
              "closed": false,
              "destination": SafeSubscriber {
                "_complete": undefined,
                "_context": Object {
                  "unsubscribe": [Function],
                },
                "_error": [Function],
                "_next": [Function],
                "_parentOrParents": null,
                "_parentSubscriber": [Circular],
                "_subscriptions": null,
                "closed": false,
                "destination": Object {
                  "closed": true,
                  "complete": [Function],
                  "error": [Function],
                  "next": [Function],
                },
                "isStopped": false,
                "syncErrorThrowable": false,
                "syncErrorThrown": false,
                "syncErrorValue": null,
              },
              "isStopped": false,
              "syncErrorThrowable": true,
              "syncErrorThrown": false,
              "syncErrorValue": null,
            },
            "isStopped": false,
            "project": [Function],
            "syncErrorThrowable": true,
            "syncErrorThrown": false,
            "syncErrorValue": null,
            "thisArg": [Circular],
          },
        ],
        "thrownError": null,
      },
    },
    "worstQuotes$": Observable {
      "_isScalar": false,
      "operator": MapOperator {
        "project": [Function],
        "thisArg": undefined,
      },
      "source": Observable {
        "_isScalar": false,
        "source": BehaviorSubject {
          "_isScalar": false,
          "_value": Array [
            Object {
              "author": "bardzo",
              "id": 0,
              "quotation": "asddasdasads adsasd a dsads  a ddsa asd.",
              "rate": -3,
            },
            Object {
              "author": "madry",
              "id": 1,
              "quotation": "AEDRFDHSRHTSHDERTaSER EAWGAERWEWF WEEF.",
              "rate": 3,
            },
            Object {
              "author": "autor",
              "id": 2,
              "quotation": "039284120934 012 4012981284 0914 29810 928.",
              "rate": 4,
            },
            Object {
              "author": "asgwfewefwd",
              "id": 3,
              "quotation": "9832 jhfp934h4r2  309 r48h23  984rh3  q98 4rhq329r3h4r98 h23r908q2h3r98h98h.",
              "rate": -5,
            },
          ],
          "closed": false,
          "hasError": false,
          "isStopped": false,
          "observers": Array [
            MapSubscriber {
              "_parentOrParents": Subscriber {
                "_parentOrParents": null,
                "_subscriptions": Array [
                  [Circular],
                ],
                "closed": false,
                "destination": SafeSubscriber {
                  "_complete": undefined,
                  "_context": Object {
                    "unsubscribe": [Function],
                  },
                  "_error": [Function],
                  "_next": [Function],
                  "_parentOrParents": null,
                  "_parentSubscriber": [Circular],
                  "_subscriptions": null,
                  "closed": false,
                  "destination": Object {
                    "closed": true,
                    "complete": [Function],
                    "error": [Function],
                    "next": [Function],
                  },
                  "isStopped": false,
                  "syncErrorThrowable": false,
                  "syncErrorThrown": false,
                  "syncErrorValue": null,
                },
                "isStopped": false,
                "syncErrorThrowable": true,
                "syncErrorThrown": false,
                "syncErrorValue": null,
              },
              "_subscriptions": Array [
                SubjectSubscription {
                  "_parentOrParents": [Circular],
                  "_subscriptions": null,
                  "closed": false,
                  "subject": [Circular],
                  "subscriber": [Circular],
                },
              ],
              "closed": false,
              "count": 1,
              "destination": Subscriber {
                "_parentOrParents": null,
                "_subscriptions": Array [
                  [Circular],
                ],
                "closed": false,
                "destination": SafeSubscriber {
                  "_complete": undefined,
                  "_context": Object {
                    "unsubscribe": [Function],
                  },
                  "_error": [Function],
                  "_next": [Function],
                  "_parentOrParents": null,
                  "_parentSubscriber": [Circular],
                  "_subscriptions": null,
                  "closed": false,
                  "destination": Object {
                    "closed": true,
                    "complete": [Function],
                    "error": [Function],
                    "next": [Function],
                  },
                  "isStopped": false,
                  "syncErrorThrowable": false,
                  "syncErrorThrown": false,
                  "syncErrorValue": null,
                },
                "isStopped": false,
                "syncErrorThrowable": true,
                "syncErrorThrown": false,
                "syncErrorValue": null,
              },
              "isStopped": false,
              "project": [Function],
              "syncErrorThrowable": true,
              "syncErrorThrown": false,
              "syncErrorValue": null,
              "thisArg": [Circular],
            },
          ],
          "thrownError": null,
        },
      },
    },
  },
}
`;
